mobile-app/app/screens/AppNavigator/screens/Balances/components/CreateOrEditAddressLabelForm.tsx

```diff
@@ -28,57 +36,127 @@ export const CreateOrEditAddressLabelForm = memo(({ route, navigation }: Props):
+  const [addressInputErrorMessage, setAddressInputErrorMessage] = useState('')
+  const validateLabelInput = (input: string): boolean => {
     if (input !== undefined && input.trim().length > 30) {
-      setInputErrorMessage('Address label is too long (max 30 characters)')
+      setLabelInputErrorMessage('Address label is too long (max 30 characters)')
+      return false
+    }
+    if (type === 'create' && input.trim() === '') {
+      setLabelInputErrorMessage('Please enter an address label')
+      return false
+    }
+    setLabelInputErrorMessage('')
+    return true
+  }
+
+  const validateAddressInput = (input: string): boolean => {
+    const decodedAddress = fromAddress(input, networkName)
+    if (decodedAddress === undefined) {
+      setAddressInputErrorMessage('Please enter a valid address')
+      return false
+    }
+    if (addressBook?.[input.trim()] !== undefined) {
+      setAddressInputErrorMessage('Please enter a unique address')
       return false
     }
-    setInputErrorMessage('')
+    setAddressInputErrorMessage('')
     return true
   }
+
   const handleSubmit = async (): Promise<void> => {
-    if (labelInput === undefined) {
-      return
+    if (type === 'edit') {
+      handleEditSubmit()
+    } else {
+      handleCreateSubmit()
     }
+  }
 
-    if (!validateInput(labelInput)) {
+  const handleEditSubmit = (): void => {
+    if (labelInput === undefined ||
+      address === undefined ||
+      !validateLabelInput(labelInput)) {
       return
     }
-
-    await onSubmitButtonPress({
+    onSaveButtonPress({
       [address]: {
         label: labelInput.trim(),
         isMine: true
       }
     })
   }
 
+  // Passcode prompt on create
+  const dispatch = useDispatch()
+  const { data: { type: encryptionType } } = useWalletNodeContext()
+  const isEncrypted = encryptionType === 'MNEMONIC_ENCRYPTED'
+  const logger = useLogger()
+  const handleCreateSubmit = useCallback(() => {
+    if (!isEncrypted ||
+      addressInput === undefined ||
+      labelInput === undefined ||
+      !validateLabelInput(labelInput) ||
+      !validateAddressInput(addressInput)) {
+      return
+    }
+
+    const auth: Authentication<string[]> = {
+      consume: async passphrase => await MnemonicStorage.get(passphrase),
+      onAuthenticated: async () => {
+        onSaveButtonPress({
+          [addressInput]: {
+            label: labelInput.trim(),
+            isMine: false
+          }
+        })
+      },
+      onError: e => logger.error(e),
+      message: translate('screens/Settings', 'Enter passcode to continue'),
+      loading: translate('screens/Settings', 'Verifying access')
+    }
+    dispatch(authentication.actions.prompt(auth))
+  }, [dispatch, isEncrypted, addressInput, labelInput, onSaveButtonPress])
+
+  const isSaveDisabled = (): boolean => {
+    if (type === 'create' && (addressInput === undefined || labelInput === undefined || labelInputErrorMessage !== '' || addressInputErrorMessage !== '')) {
```

Comment:
instead of using type, should we use a boolean? Since there are only two states, `create` or `delete`

Reply:
There is a future use case for "create wallet address" in wallet address bottom sheet in the future. 
And it has different requirement compare to create new address for address book, e.g.: label field is optional there vs mandatory here

Reply:
Possible types:
1. edit address label (wallet address)
2. new address created (wallet address)
3. add new address (address book)
4. edit address (address book)

But instead of separate by `create` or `edit`, i think it would be better as `addressBook` and `walletAddress`, but since you prefer boolean, then i'll go with `isAddressBook`?

Reply:
Ah ok understand. Ok we can use the `addressBook` and `walletAddress` instead of boolean.

Reply:
But if we move this to a dedicated screen, would it make sense to change this type?